### ISSUE TITLE
"Choose column to display" does not work

### ISSUE THREAD

**Issue by svivian** [phpmyadmin/phpmyadmin#13533]
### Steps to reproduce
1. Go to a table view.
2. Go to the Relations tab.
3. Next to "Choose columns to display", pick another column.

### Expected behaviour

The column to display should change.

### Actual behaviour

Firstly, the dropdown started with a field selected that I don't remember selecting in the first place, and didn't want. So I tried to change it to something useful. Clicking save appears to do something (it pops up the "Loading" banner) but it doesn't change the field. If I click "Preview SQL" it always says "No change".

If I'm not mistaken the preferences should be stored in the table `phpmyadmin.pma__table_info` which I have set up correctly (other functions like query history are working fine). That table did have some config for other databases in there that I added a while back, but nothing was getting added to it. Tried emptying the table but still no joy.

### Server configuration
**Operating system**: Mac

**Web server:** Apache

**Database:** Mysql 5.6.13

**PHP version:** 5.5.38

**phpMyAdmin version:** 4.7.3

### Client configuration
**Browser:** Chrome

**Operating system:** Mac

**Comment by ibennetch** [comment]
Indeed, this also seems broken on master.

**Comment by godzillante** [comment]
I can confirm it happens with other browsers on Mac as well (Firefox 54 and Safari 10)

**Comment by jebsolutions** [comment]
How to best fix this is left up to your imagination.  My quick hack was to do an "insert...on duplicate key" sql instead of an "update":

```

    public function updateDisplayField($disp, $display_field, $cfgRelation)
    {
        $upd_query = false;
        if ($disp) {
            if ($display_field == '') {
                $upd_query = 'DELETE FROM '
                    . Util::backquote($GLOBALS['cfgRelation']['db'])
                    . '.' . Util::backquote($cfgRelation['table_info'])
                    . ' WHERE db_name  = \''
                    . $GLOBALS['dbi']->escapeString($this->_db_name) . '\''
                    . ' AND table_name = \''
                    . $GLOBALS['dbi']->escapeString($this->_name) . '\'';
            } elseif ($disp != $display_field) {
                $upd_query = 'INSERT INTO '
                    . Util::backquote($GLOBALS['cfgRelation']['db'])
                    . '.' . Util::backquote($cfgRelation['table_info'])
                    . '(db_name, table_name, display_field) VALUES('
                    . '\'' . $GLOBALS['dbi']->escapeString($this->_db_name) . '\','
                    . '\'' . $GLOBALS['dbi']->escapeString($this->_name) . '\','
                    . '\'' . $GLOBALS['dbi']->escapeString($display_field) . '\')'
                    . ' ON DUPLICATE KEY UPDATE '
                    . ' display_field = \''
                    . $GLOBALS['dbi']->escapeString($display_field) . '\'';
            }

```


I disclaim all right to the above code.  anybody can use it for any reason.

**The problem is caused by this code**:

https://github.com/phpmyadmin/phpmyadmin/blame/b2ad33ca474ba2c3355d8e118703331209d10f3d/libraries/relation.lib.php#L867

If there is no "display_field" set for this table it defaults to the the first character field.

**And in this function**
https://github.com/phpmyadmin/phpmyadmin/blob/2ff392043820ce372a539ca2e1afdde5ad9f884c/libraries/classes/Table.php

public function updateDisplayField($disp, $display_field, $cfgRelation)

It compares $disp (which can be the defaulted value) with your changed value (whatever you set).  Since those two do not match...it does an UPDATE instead of an INSERT.  

And an update to a non-existent key "works" but does nothing since there is no record to overwrite....so thats why there is no sql error but the display field wasn't being updated.


anyway.  that's all the patience I have for this issue, someone else can do a PR.

**Comment by nijel** [comment]
@jebsolutions Thanks for analysis, I've just fixed that in ed912090ec3f7840f6f9c0b20ed22af56d9f9bc1.

**Comment by nijel** [comment]
It might be cached in the session as well (I'm not sure with this, but AFAIK there was something like this).
